### ISSUE TITLE
TypeError: Cannot read property '$options' of undefined - After Nuxt JS upgrade to 2.14.12 from 2.14.6

### ISSUE THREAD

**Issue by PayelKarmakar06** [nuxt/nuxt.js#8600]
Previous Nuxt version 2.14.6
Upgrade version 2.14.12

**mode: 'spa'** has been modified into **ssr: false**,
Error in console -

TypeError: Cannot read property '$options' of undefined
    at Object.get (app.js:2495)
    at Store.initialize (app.js:18822)
    at Array.wrappedActionHandler (app.js:27261)
    at Store.dispatch (app.js:26926)
    at Store.boundDispatch [as dispatch] (app.js:26816)
    at _callee$ (app.js:8353)
    at tryCatch (app.js:8624)
    at Generator.invoke [as _invoke] (app.js:8858)
    at Generator.prototype.<computed> [as next] (app.js:8676)
    at asyncGeneratorStep (app.js:91)

Error being thrown for the below piece of code in /_nuxt/app.js

get: function get() {
    return this.$root.$options.$nuxt;
  },

**What is Expected?**
Working like version v2.14.6

**What is actually happening?**
TypeError

**Comment by PayelKarmakar06** [comment]
I don't have the privilege to do that. Can you please ask for some relevant info I can help you with @Atinux ?

Do we need to modify any specific configuration in the nuxt.config file after the recent upgrades? 

P.S I have deleted yarn.lock and node_modules folder, and then did a reinstall it did not help.

**Comment by PayelKarmakar06** [comment]
@danielroe I checked the sandbox, and it indeed does not have the issue I am facing.

So int the nux.config I am having -- 

const options = {
    srcDir: './src',
    buildDir: './.nuxt',
    buildModules: [
            '@nuxtjs/vuetify',
        ],
    build: {
            analyze: false,
            cache: true,
    },
    modules: ['nuxt-i18n' ],
    vuetify: {
           treeShake: true,
     }
}
module.exports = options

After build when the .nuxt directory is generated, in the App.js folder this.$root is not available -

get: function get() {
    return this.$root.$options.$nuxt;
  },

Hence I am getting the above mentioned error.

I tried replacing the instances of Vue.prototype.nuxt by this.$nuxt, which is being used in locale.js file - but this.$nuxt is also not available.
